probes/webhooksWithoutTokenAuth/impl.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package webhooksWithoutTokenAuth
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "webhooksWithoutTokenAuth"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	r := raw.WebhookResults
+
+	var findings []finding.Finding
+	for _, hook := range r.Webhooks {
+		if !hook.UsesAuthSecret {
```

Comment:
since there are multiple webhooks, I think we want the code to be able to support that. I would crate a positive / negative error based on the value of `hook.UsesAuthSecret` in each iteration. If the `len(r.Webhooks) == 0`, then there's no webhook and we could return either positive or notApplicable

Reply:
This is related to https://github.com/ossf/scorecard/pull/3522#discussion_r1338967364. Will adjust.

Reply:
This is also related to https://github.com/ossf/scorecard/pull/3522#discussion_r1338954229, so leaving it until that discussion has resolved.

Reply:
Fixed in This should no longer be relevant because of https://github.com/ossf/scorecard/pull/3522#issuecomment-1811514146